### ISSUE TITLE
Question about assignment3

### ISSUE THREAD

**Issue by bk820653874** [cuhkrlcourse/ierg6130-assignment#7]
In assignment3, is the output of net.forward() F.softmax(x)? If the return is x, logit may not be a distribution when the output has both positive and negative components.

**Comment by pengzhenghao** [comment]
In the real practice, we usually use a the raw output of the last layer and then add an extra `distribution` class on it. So the output of the network is simply a raw output of a FC layer. No activation, no softmax.

Why we do that? Because not like in classification task and regression task that the loss can be compute easily by doing some math on the output of the network, in RL we usually need to compute other values BASED ON the original output of the network, like KL divergence and entropy of the action distribution.

For example, if you are in a continuous control task and wish to output a beautiful Gaussian distribution, what should you do? Say the action dimension is K and you need to output 2K values. You need to first split the 2K values into a `mean` vector (length-K) and a `std` vector (length-k). Then compute the KL or entropy based on these two vector.

So the structure of the MLP becomes quite complex. So generality, people tend to use a MLP that return whatever number of values as output and then add an extra `distribution` on it.

In short, the output of `net.forward` is not necessary to be softmax. We conduct the softmax operation later to get the real probability.

**Comment by bk820653874** [comment]
Thanks you very much.
